src/shared/proto/robot_component_status.proto

```diff
@@ -0,0 +1,176 @@
+    optional bool autokick_fired = 8;
+}
+
+/* Data from the encoder */
+message EncoderData
+{
+    optional int32 motor_fl_speed = 1;
+    optional int32 motor_fr_speed = 2;
+    optional int32 motor_bl_speed = 3;
+    optional int32 motor_br_speed = 4;
+}
+
+/* Error codes are to be displayed in the visualizer */
+enum ErrorCode
+{
+    /** 2014 ROBOT Messages **/
+    // Electrical messages
+    ROBOT_DEAD_MESSAGE      = 1;
+    LOW_CAP_MESSAGE         = 2;
+    LOW_BATTERY_MESSAGE     = 3;
+    HIGH_BOARD_TEMP_MESSAGE = 4;
+    // Level triggered events
+    CHARGE_TIMEOUT                  = 5;
+    WHEEL_0_MOTOR_HOT               = 6;
+    WHEEL_1_MOTOR_HOT               = 7;
+    WHEEL_2_MOTOR_HOT               = 8;
+    WHEEL_3_MOTOR_HOT               = 9;
+    DRIBBLER_MOTOR_HOT              = 10;
+    WHEEL_0_ENCODER_NOT_COMMUTATING = 11;
+    WHEEL_1_ENCODER_NOT_COMMUTATING = 12;
+    WHEEL_2_ENCODER_NOT_COMMUTATING = 13;
+    WHEEL_3_ENCODER_NOT_COMMUTATING = 14;
+    WHEEL_0_HALL_SENSOR_STUCK_LOW   = 15;
+    WHEEL_1_HALL_SENSOR_STUCK_LOW   = 16;
+    WHEEL_2_HALL_SENSOR_STUCK_LOW   = 17;
+    WHEEL_3_HALL_SENSOR_STUCK_LOW   = 18;
+    DRIBBLER_HALL_SENSOR_STUCK_LOW  = 19;
+    WHEEL_0_HALL_SENSOR_STUCK_HIGH  = 20;
+    WHEEL_1_HALL_SENSOR_STUCK_HIGH  = 21;
+    WHEEL_2_HALL_SENSOR_STUCK_HIGH  = 22;
+    WHEEL_3_HALL_SENSOR_STUCK_HIGH  = 23;
+    DRIBBLER_HALL_SENSOR_STUCK_HIGH = 24;
+    // Edge triggered events
+    ICB_CRC_ERROR                        = 25;
+    RECEIVE_FRAME_CHECK_SEQUENCE_FAILURE = 26;
+    CRASHED_CORE_DUMPED                  = 27;
+    CRASHED_NO_CORE_DUMP                 = 28;
+
+    /** 2014 DONGLE Messages **/
+    ESTOP_BROKEN_MESSAGE        = 29;
+    RX_FCS_FAIL_MESSAGE         = 30;
+    SECOND_DONGLE_MESSAGE       = 31;
+    TRANSMIT_QUEUE_FULL_MESSAGE = 32;
+    RECEIVE_QUEUE_FULL_MESSAGE  = 33;
+}
+
+/* Firmware build information and primitive status  */
+message FirmwareStatus
+{
+    // The microcontroller firmware build ID.
+    required uint32 fw_build_id = 1;
+
+    // Whether the robot is in direct mode.
+    required bool direct_control = 2;
+
+    // Whether or not the build ID information is valid.
+    required bool build_ids_valid = 3;
+
+    /** 2014 robots-specific fields**/
+    // The FPGA bitstream build ID.
+    optional uint32 fpga_build_id = 4;
+}
+
+/* Data about the status of the motors */
+message MotorStatus
+{
+    // The speed of the robot’s dribbler motor, in revolutions per minute.
+    required int32 dribbler_speed = 1;
+
+    // The maximum power level understood by the direct_dribbler function.
+    required uint32 direct_dribbler_max = 2;
+
+    /* Frankie Robot Fields */
+    // Encoder information
+    optional EncoderData encoder_data = 3;
+}
+
+/* Data about the network connection with the robots, including network-derived values */
+message NetworkStatus
+{
+    // When the proto was created according to lwIP SNTP time
+    required uint64 time_created = 1;
```

Comment:
@akhilveeraghanta I'm assuming this field is useful for when we do filtering to get a timestep / timestamp for new values like encoders? Just checking because otherwise not sure how useful this is.

Reply:
When Jon and I were discussing how we detect timeouts and get better metrics on latency, we thought including the time_created in the packet because lwip supports sntp is a really good diagnostic value to get that sort of information. The time would be synced up between the STM32 and the AI because they are on the same network.

We could also use it for filtering, but its mainly just for diagnostics